From patchwork Tue Sep  6 19:48:41 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Liam Howlett <Liam.Howlett@Oracle.com>
X-Patchwork-Id: 12968213
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 117ECECAAA1
	for <linux-mm@archiver.kernel.org>; Tue,  6 Sep 2022 21:27:16 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id 5939D8D0005; Tue,  6 Sep 2022 17:27:16 -0400 (EDT)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id 542A78D0002; Tue,  6 Sep 2022 17:27:16 -0400 (EDT)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id 394CD8D0005; Tue,  6 Sep 2022 17:27:16 -0400 (EDT)
X-Delivered-To: linux-mm@kvack.org
Received: from relay.hostedemail.com (smtprelay0015.hostedemail.com
 [216.40.44.15])
	by kanga.kvack.org (Postfix) with ESMTP id 29B578D0002
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 17:27:16 -0400 (EDT)
Received: from smtpin23.hostedemail.com (a10.router.float.18 [10.200.18.1])
	by unirelay01.hostedemail.com (Postfix) with ESMTP id F14FC1C2C65
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 21:27:15 +0000 (UTC)
X-FDA: 79882946430.23.916D514
Received: from mx0a-00069f02.pphosted.com (mx0a-00069f02.pphosted.com
 [205.220.165.32])
	by imf09.hostedemail.com (Postfix) with ESMTP id 8ACFC14007C
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 21:27:14 +0000 (UTC)
Received: from pps.filterd (m0246629.ppops.net [127.0.0.1])
	by mx0b-00069f02.pphosted.com (8.17.1.5/8.17.1.5) with ESMTP id
 286IdAiV024665;
	Tue, 6 Sep 2022 19:48:46 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : references : in-reply-to : content-type :
 content-transfer-encoding : mime-version; s=corp-2022-7-12;
 bh=40Zfz471W6ukNOyhk0Qf2GGR9Z4MieDkOVdFAcoHHpI=;
 b=FmMp/qj+SUlGyPjC7AxkYLNEGIS5X72t7VEdY52frUlVnGLi0nljO856l2N4tYEsxTpi
 PlywHomt5MVjKbZgSfoO9ICoSYs254sD/6tWx0zfmoSNjOf7IVDeu+CfRqEVvZL56HSk
 p5IXxrUCzSiCQdZkMWdtKVCbu9E44D7hiTYbw7jpDOeYUuy7ItMVgOGHEZ0BlpaI4Zdw
 bmz+8Z8EbZDo6uuRaGeEW2Hlo8ztcuaVtsMKYJIjVp1eCLsmPH3cWl3d31jjKfyV3Wif
 TcCpCB06/K3K7VswHsxMpLckE/+/oW2RY8yZ17Z91+Du+X0H5I1FVWQN1s1gD5e2kUex 8A==
Received: from phxpaimrmta02.imrmtpd1.prodappphxaev1.oraclevcn.com
 (phxpaimrmta02.appoci.oracle.com [147.154.114.232])
	by mx0b-00069f02.pphosted.com (PPS) with ESMTPS id 3jbxtaet4s-1
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:48:46 +0000
Received: from pps.filterd
 (phxpaimrmta02.imrmtpd1.prodappphxaev1.oraclevcn.com [127.0.0.1])
	by phxpaimrmta02.imrmtpd1.prodappphxaev1.oraclevcn.com (8.17.1.5/8.17.1.5)
 with ESMTP id 286HHdKM037489;
	Tue, 6 Sep 2022 19:48:45 GMT
Received: from nam12-bn8-obe.outbound.protection.outlook.com
 (mail-bn8nam12lp2171.outbound.protection.outlook.com [104.47.55.171])
	by phxpaimrmta02.imrmtpd1.prodappphxaev1.oraclevcn.com (PPS) with ESMTPS id
 3jbwc32vns-5
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:48:45 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=hTyJTHZoZmvULpQvv9p1RfrZIPXawQjfW342bYFT0ZlU3XS05u2P5JZU2RqpouFo7MWqnFyGi28a0hXeGxA7LABNes9uHtgvlq/NmDUYpPYT/wN69oLA+Bs/Ga6RyONd0OiLcMN78RRt7i+MHrMNvukK102OP5atnB9qnsWF7rWr3yIWLfWuX/h1XajceEBSSBgrr44IJ3bhb2RXS+eMaGR2ym+JY48WJPt/xF2NODr0RT7AIqHXcwtYOUfngZGdlaeNjdShGNWRucfpeohdINioyhuFd0zdoCC9RAWp9q6yVdCGAIowoQo7t1Dw3t+pkHeCHkmmpBEw0TSeqTjviQ==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=40Zfz471W6ukNOyhk0Qf2GGR9Z4MieDkOVdFAcoHHpI=;
 b=kH2ErPcOKyFTCJWfYooe8XUn1hYkc5+bCFyvWLGKeu4A+FA/IlRjovcGB7SRukQH1TZYPCpLya/4pvrMZa04rnzwE4ijkAVlvJbXhTT7beBPgtrCrhq50TdqgmQc7x4VuRi3e+NTOp0Dih34cqTnIGDuD3ns3oc5PUYYOn6yQbKYVK0FykIhBs4ZlLgN2tNePFVbAXF9IHFhymCXAzikt8+IAXloACh+agK4ggNlkeOqgbggx3j/MVrBdqKg82MPkxvVspbf67PSoLwDe5I5BDFp8eAYVI4qE5UozBB3nH5vAgDEjx+ib6TlnLPH51QoR6K3wrg9UjUoZiS0CtkgNQ==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=40Zfz471W6ukNOyhk0Qf2GGR9Z4MieDkOVdFAcoHHpI=;
 b=ma+6m3S72e+1KNeu1p6xU/LJaQLxATQw932ZrhAe8AGridxzr5f6tuavKx5CFNMfivZupANvb8obKVvvmDSeljCrqgYDAytG1IpXcsnBwof9hQI1f/trOlEpLonnkUTX8QJrcKlJkWFPe2ymXjK1BycMGfTEX3yopOm18rCeuAw=
Received: from SN6PR10MB3022.namprd10.prod.outlook.com (2603:10b6:805:d8::25)
 by DM4PR10MB6183.namprd10.prod.outlook.com (2603:10b6:8:8b::22) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.5588.10; Tue, 6 Sep
 2022 19:48:42 +0000
Received: from SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223]) by SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223%5]) with mapi id 15.20.5588.018; Tue, 6 Sep 2022
 19:48:42 +0000
From: Liam Howlett <liam.howlett@oracle.com>
To: "maple-tree@lists.infradead.org" <maple-tree@lists.infradead.org>,
        "linux-mm@kvack.org" <linux-mm@kvack.org>,
        "linux-kernel@vger.kernel.org"
	<linux-kernel@vger.kernel.org>,
        Andrew Morton <akpm@linux-foundation.org>
CC: Liam Howlett <liam.howlett@oracle.com>,
        Liam Howlett
	<liam.howlett@oracle.com>
Subject: [PATCH v14 06/70] radix tree test suite: add lockdep_is_held to
 header
Thread-Topic: [PATCH v14 06/70] radix tree test suite: add lockdep_is_held to
 header
Thread-Index: AQHYwimqxXLoN4eu0Ua0eRnV+uwFyg==
Date: Tue, 6 Sep 2022 19:48:41 +0000
Message-ID: <20220906194824.2110408-7-Liam.Howlett@oracle.com>
References: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
In-Reply-To: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
Accept-Language: en-US
Content-Language: en-US
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
x-mailer: git-send-email 2.35.1
x-ms-publictraffictype: Email
x-ms-office365-filtering-correlation-id: 8cc36eb6-2db6-4143-9574-08da9040cd30
x-ms-traffictypediagnostic: DM4PR10MB6183:EE_
x-ms-exchange-senderadcheck: 1
x-ms-exchange-antispam-relay: 0
x-microsoft-antispam: BCL:0;
x-microsoft-antispam-message-info: 
 UqHuYGXQvDBTkI7zoLd1KkWjRIppVzCl0W3lg/GwAYMJEL+2Drw+BqHXpELBD5aT+l3gEvPYkrQVhgVNF9YF93X6iveWrPi+FVRkCac2N2CQu4tz/hn6pxVn01zJEzlv1IjM6fusxbjSGP9MusJpayqKmtrhByeRxqZg16WBMV1kjHIxd4W87j00z3OUF3m50yFF5XoVFXcp0eUeNkiKIK63ckilf5bdCrN6lc4/l6i+GcEiifwuc1S9FoVYozqi7a2PkqrKAe5kMy94Ka5B4bw1XEHEs4dEJgQyYiwIN5TkH/hY53bfvZa6T0t/2eHjv4WRAmGnPhpnwCqX2zK3lhSdFMTRrZKSh7aaGe5aUKrkKzfH99/fzD4rqiPYQRqA8zfI/2DNkXUYC0p3jTvZUY8WQpj52KPhq1avpNBMasswG8JcUqjOzQd1EoABLO0hKsrKz6CEAQGrLuI6MUiBQw9Ao3WHbIH1Bo0QK0GRM+CuHm+AZEQ9tr8F7/b7TWKNMSYYYiP5/yWEhRyfZAmKiG81/rF0LGdSe52dHJfIQPRU0kmyn38ZWjLEYlApkP/iTDtrbjq0AexPVg4VH1sCb63LEMuC8P4q8vsKbKkC0Fagk3m8RUWFbUJUykl9xvzbte+n0frSMIjX77x++EQTAnsQuEpC0QJR5DF4PwtV6wTwhMRi7Phz+82UZy6OWYrz4qw+IHaHG/K+6yAuh+OZyxNoZ0l4bBzk53ToLdDn9TU7p24sd3CSHK3befd3r+M1faznPfCfKernDy6/xJYe/Q==
x-forefront-antispam-report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3022.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(13230016)(39860400002)(366004)(376002)(396003)(346002)(136003)(66556008)(38100700002)(122000001)(8936002)(5660300002)(76116006)(66946007)(66446008)(66476007)(2906002)(4744005)(83380400001)(8676002)(91956017)(36756003)(2616005)(186003)(107886003)(86362001)(1076003)(110136005)(44832011)(71200400001)(38070700005)(54906003)(26005)(64756008)(41300700001)(6506007)(6512007)(6486002)(316002)(478600001)(4326008);DIR:OUT;SFP:1101;
x-ms-exchange-antispam-messagedata-chunkcount: 1
x-ms-exchange-antispam-messagedata-0: =?iso-8859-1?q?JG5n1z9WxqNUDaJ9THJM+Or?=
	=?iso-8859-1?q?/nZO6kIZLISxEEYTKunYWHM+M/NSkqUGgc26+uEIHveK6Jnvgw1D8yMULk6W?=
	=?iso-8859-1?q?9O2yANnUNnhHRpGnE5H/zM5RiINF87YpBkjhbbJ0I91sY1id5RmmHYsYNJuA?=
	=?iso-8859-1?q?sIjK51GGlozaQIJEs0AW4rYq3m14OItsx8xob0NbEE0G/mCGcmvyPprUt4aN?=
	=?iso-8859-1?q?4t1/iXICog2x3pGk6s6ku6Y0Jrvv7GGtTitSyDQaVtBpwsIbl8E6emJCjCFG?=
	=?iso-8859-1?q?Pk3MqfAOMFKv9zCNrnjCrwvx/8Z7s07mUpzKMZXCW3JtIyRpByepTQphSdEH?=
	=?iso-8859-1?q?ioVwa5HTM/2hnT6zBHlQDtpV18uqfLDkDW1T7YQ2xR1A7McTSVP+EADnwuMA?=
	=?iso-8859-1?q?Pv4kOvbTJ3INkuc07sjzVh/VFwtE3ntX3R3HCe19p+ZLYkpjahzPZsvh7iPA?=
	=?iso-8859-1?q?cOpZyuXp/ztRC3H8shZdObTWOGHjUe6QMqUv1Lb1jFxIEYUdekElhW7H5/Vb?=
	=?iso-8859-1?q?/HG21fZTvGAzzZIc16yxBL6O3X33IlRTG9809ywhtOTaHa2tgXYdgcsD8Dby?=
	=?iso-8859-1?q?exDe29P4FRCRU5TLABLNY5UTLWYb8YWahm6iIjKOhuVYGItP8zitRZUVYduj?=
	=?iso-8859-1?q?KUsLMihl0VpTtW6qOdMrLBTXd8wv5ckL2F45xOSsCzAcDfSPvILwy5R0qCFD?=
	=?iso-8859-1?q?OVaCi+7ZoNwl3I1YBSZdWS5BHjzE+oSb78cjF5uL0GfMTu4DmDyb23FtAWEx?=
	=?iso-8859-1?q?Pd03m9fFU4BHH0oZYiVZG/I3U0upkE8lgzoyOwJO3TbX0XLVfRwNKUVWaL9/?=
	=?iso-8859-1?q?njOJJ4Aaz3I3EtlMSfyN70y3pingRGj14fYFu/U3I0NI3c7FT7qf8oYpSkEm?=
	=?iso-8859-1?q?mrjtSNuiVwWM1DE9LNLBHkzEkGeP9Mmr1h29meXqg6JLFNtE0OCMekvVS2cw?=
	=?iso-8859-1?q?UfDE9pZmUdpmMIkEZzfWyiN8IBi3I9+P0XWl61aTn4vLqiVPpY3KGn0izIRm?=
	=?iso-8859-1?q?dj7CbyJDfAI1MePChBZSctQmsMqllC7sYVQA1Fs0u/w/38ILqz8bxVfDXr2S?=
	=?iso-8859-1?q?gAo2jDLQiipktvDwzC6R8V7DFDkFoa5RgEysra21kg32Fb9PObcJTm3xIsY9?=
	=?iso-8859-1?q?WE4/0scvvL0ftcuNzh6NqmSHSi0XD2IHC1LTtn2482DjeJkL1qchwXKLUAqD?=
	=?iso-8859-1?q?TT2JlS2wVyr69IOwLuUw9R+pb3xjPxk6dB2I3b2gaR3Q+JCgyGJQJrAQAHpk?=
	=?iso-8859-1?q?yZBn1+ncxaCCZKGl8ixvUjUKxVOTtouZMSmUkFGTem8n1FKeboo/iASFEN/L?=
	=?iso-8859-1?q?tMXAU5AZwtqZ2amgrA+L2HZCvUAIf81jGssOEwacgTs+hWFTrCEZzn1TlM0x?=
	=?iso-8859-1?q?E9iYCAhgrw052yji/JjbZrwdjCHY6CuvmRQTD0mpo+BJTLqVCSWkrOtoV9D0?=
	=?iso-8859-1?q?I3aYKbRKQitzXytH6aTHV67aeDJ2EOUVY4+RtVdhdCLeBKKh2EfL+02Lmnwx?=
	=?iso-8859-1?q?aSmn1lV2EKbqPMKTt4CnxuDYXI7gy8PD5QnpQJQqXBYT2iN9ZddGOYYs0JWL?=
	=?iso-8859-1?q?UJ3aLLoO5/boL6CuJfBWhp3HvayT47MEupXLYZOzRs3XZIKHqnp5t1aJ9W9d?=
	=?iso-8859-1?q?02wih1O9pz7VqEwDOGNHIFCFqskXxtPY3j1klGA=3D=3D?=
MIME-Version: 1.0
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3022.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 8cc36eb6-2db6-4143-9574-08da9040cd30
X-MS-Exchange-CrossTenant-originalarrivaltime: 06 Sep 2022 19:48:41.5541
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: 
 gE24YrFvpJczvjujgp4qF/ZtRsBAM1pW1fXcYEcTTUESBRBVwdkU3q57e6qd/nj6LO0DKt+mekc62Iq2ETbL7A==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: DM4PR10MB6183
X-Proofpoint-Virus-Version: vendor=baseguard
 engine=ICAP:2.0.205,Aquarius:18.0.895,Hydra:6.0.528,FMLib:17.11.122.1
 definitions=2022-09-06_09,2022-09-06_02,2022-06-22_01
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 bulkscore=0
 suspectscore=0 phishscore=0
 mlxlogscore=999 spamscore=0 adultscore=0 mlxscore=0 malwarescore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2207270000
 definitions=main-2209060091
X-Proofpoint-ORIG-GUID: SdlIL4q8KswUJkpUpavYB_afvG8ZzwQi
X-Proofpoint-GUID: SdlIL4q8KswUJkpUpavYB_afvG8ZzwQi
ARC-Seal: i=2; s=arc-20220608; d=hostedemail.com; t=1662499634; a=rsa-sha256;
	cv=pass;
	b=0zGpuLP7Q/2wxK++jPuqjSZ8n3C/HEmaOPCGn8t0bT3k85kJKjNxrU/iKakAnEH0MDNvoY
	fkMJnp1u+QLVJ0fRNXzHBkd6u8AYTC/kwCAdBiKw/goEYST4AXAcvPv8njCKj1mXt1+9xg
	5WKXHSNHu0L7b7yrydQjcikEPpA5xfE=
ARC-Authentication-Results: i=2;
	imf09.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b="FmMp/qj+";
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b=ma+6m3S7;
	spf=pass (imf09.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.165.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com;
	arc=pass ("microsoft.com:s=arcselector9901:i=1")
ARC-Message-Signature: i=2; a=rsa-sha256; c=relaxed/relaxed;
 d=hostedemail.com;
	s=arc-20220608; t=1662499634;
	h=from:from:sender:reply-to:subject:subject:date:date:
	 message-id:message-id:to:to:cc:cc:mime-version:mime-version:
	 content-type:content-type:
	 content-transfer-encoding:content-transfer-encoding:
	 in-reply-to:in-reply-to:references:references:dkim-signature;
	bh=40Zfz471W6ukNOyhk0Qf2GGR9Z4MieDkOVdFAcoHHpI=;
	b=OwXQUlOs7wqFiVo9GSVUf2+BL8IWbOSvIzG9m4BDMtTcB2159t4GVCIJQdk2yWyFa/CeGn
	rVnalQnwAP3mL/q4o9Xj5hhVYG8tlvu35pfL3RGO4s7JVjfEf4Y3xLQNRLnwejzY+EpQRO
	jAT1Icv4nheL7Tuu1lnduRaxpHe3qsU=
X-Stat-Signature: x9onmw3hrkc73imwesdqc9i3rbzub4od
X-Rspamd-Queue-Id: 8ACFC14007C
X-Rspam-User: 
Authentication-Results: imf09.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b="FmMp/qj+";
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b=ma+6m3S7;
	spf=pass (imf09.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.165.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com;
	arc=pass ("microsoft.com:s=arcselector9901:i=1")
X-Rspamd-Server: rspam08
X-HE-Tag: 1662499634-373194
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

From: "Liam R. Howlett" <Liam.Howlett@Oracle.com>

maple tree uses lockdep_is_held, so define it as external in the header.

Signed-off-by: Liam R. Howlett <Liam.Howlett@oracle.com>
---
 tools/testing/radix-tree/linux/lockdep.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tools/testing/radix-tree/linux/lockdep.h b/tools/testing/radix-tree/linux/lockdep.h
index 016cff473cfc..62473ab57f99 100644
--- a/tools/testing/radix-tree/linux/lockdep.h
+++ b/tools/testing/radix-tree/linux/lockdep.h
@@ -11,4 +11,6 @@ static inline void lockdep_set_class(spinlock_t *lock,
 					struct lock_class_key *key)
 {
 }
+
+extern int lockdep_is_held(const void *);
 #endif /* _LINUX_LOCKDEP_H */

